frontend/src/components/Modal/Modal.stories.tsx

```diff
@@ -0,0 +1,113 @@
+import { useDisclosure } from '@chakra-ui/hooks'
+import {
+  Box,
+  ButtonGroup,
+  Modal,
+  ModalBody,
+  ModalCloseButton,
+  ModalContent,
+  ModalFooter,
+  ModalHeader,
+  ModalOverlay,
+  ModalProps,
+} from '@chakra-ui/react'
+import { DecoratorFn, Meta, Story } from '@storybook/react'
+
+import { viewports } from '~utils/storybook'
+import Button from '~components/Button'
+
+export default {
+  title: 'Components/Modal',
+  decorators: [],
+  parameters: {
+    layout: 'fullscreen',
+    // Prevent flaky tests due to modal animating in.
+    chromatic: { delay: 200 },
+  },
+} as Meta
+
+// Required for Chromatic to know the dimensions of the snapshot to take,
+// since the modal is rendered in a portal and Chromatic only detects the
+// bounding box of the button that opens the modal.
+const desktopDecorator: DecoratorFn = (storyFn) => (
+  <Box w="1000px" h="700px">
+    {storyFn()}
+  </Box>
+)
+const mobileDecorator: DecoratorFn = (storyFn) => (
+  <Box w="320px" h="568px">
+    {storyFn()}
+  </Box>
+)
+
+type StoryModalProps = ModalProps & {
+  bodyContent?: any
+}
+
+// Helper method to generate lorem text.
+const generateLorem = (numParagraph = 1) => {
+  const para =
+    'Lorem ipsum dolor sit amet, consectetur adipiscing elit. Nulla gravida nisl sit amet hendrerit elementum. Phasellus et aliquet quam, nec ornare ex. Donec et luctus justo, id consectetur nisi. Nam non nunc ligula. Suspendisse potenti. Integer accumsan sem quis purus euismod, et tincidunt velit faucibus. Curabitur pharetra finibus nunc, sed finibus ex hendrerit eu. Cras elementum dui a imperdiet porttitor. Donec vulputate hendrerit quam, quis consectetur orci interdum ac. Cras porttitor arcu lacus.'
+  let builtString = ''
+  while (numParagraph-- > 0) {
+    builtString += para + '\n\n'
+  }
+  return builtString
```

Comment:
first while loop I've ever seen you write!
```suggestion
  return Array(numParagraph).fill(para).join('\n\n')
```